tesseract_kinematics/include/tesseract_kinematics/core/utils.h

```diff
@@ -218,6 +218,20 @@ inline static bool dampedPInv(const Eigen::Ref<const Eigen::MatrixXd>& A,
   return true;
 }
 
+/**
+ * @brief Check if the provided jacobian is near a singularity
+ * @details This is keep separated from the forward kinematics because special consideration may need to be made
+ * based on the kinematics arrangement.
+ * @param jacobian The jacobian to check if near a singularity
+ * @param threshold The threshold that all singular values must be above to not be considered near a singularity
+ */
+inline bool isNearSingularity(const Eigen::Ref<const Eigen::MatrixXd>& jacobian, double threshold = 0.01)
+{
+  Eigen::JacobiSVD<Eigen::MatrixXd> svd(jacobian, Eigen::ComputeThinU | Eigen::ComputeThinV);
+  const Eigen::VectorXd& sv = svd.singularValues();
+  return !(sv.tail(1).value() > threshold);
```

Comment:
```suggestion
  return sv.tail(1).value() < threshold;
```

Inverting the logic here might make this a little more understandable and match your documentation description

Reply:
The descriptions states `The singular values must be greater than the threshold` and the suggested change would mean `The singular values must be greater than or equal to the threshold`. I am good with the change just want to make sure we are on the same page and I will update the documentation.

Reply:
@marip8 I made the change and updated the documentation to reflect the behavior.